src/templates/_special/async-csrf-input.twig

```diff
@@ -1,19 +1,39 @@
 <script>
   (function() {
-    fetch('{{ url }}', {
-      headers: {
-        'Accept': 'application/json',
+    function fetchCsrfData() {
+      return fetch('{{ url }}', {
+        headers: {
+          'Accept': 'application/json',
+        },
+      }).then((response) => response.json());
+    }
+
+    function createHiddenInput({name, value}) {
+      const input = document.createElement('input');
+      input.type = 'hidden';
+      input.name = name;
+      input.value = value;
+      return input;
+    }
+
+    function handleFocusIn(e) {
+      const asyncInput = e.currentTarget.querySelector('craft-csrf-input');
+      if (asyncInput) {
+        fetchCsrfData().then(({csrfTokenName, csrfTokenValue}) => {
+          const input = createHiddenInput({name: csrfTokenName, value: csrfTokenValue});
+          asyncInput.replaceWith(input);
+        });
```

Comment:
I think we also need to defer/debounce submit events that come in between `focusin` and injecting the input.

Over a slow connection, you could interact with the form and submit before the CSRF has been injected, resulting in a CSRF error.